### PR TITLE
proposal-regexp-modifiers: Tests for syntax errors

### PR DESCRIPTION
**Pull request by cjtenny**
This covers the syntax error cases from https://github.com/tc39/test262/issues/3756 . Recommend reviewing commit-by-commit, since the third commit is just the result of running the python script to generate the cases. The second commit includes a few non-generated cases written by hand.

### CODE REVIEW

**Review by gibson042 — CHANGES_REQUESTED** [review]
Thanks, this is _excellent_ coverage! A few nits, but I like the overall approach.

File: src/regexp-modifiers/add-remove-multi-duplicate.case [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  - flags-syntax-error/arithmetic-modifiers-literal-reversed.template
+  - flags-syntax-error/arithmetic-modifiers-constructed-reversed.template
+desc: It is a Syntax Error if the any code point in the source text matched by the first RegularExpressionFlags is also contained in the source text matched by the second RegularExpressionFlags.
+---*/
+
+//- add-modifiers
+ims
+//- global-modifiers
+//- modifiers
+m
```
[the remark this answers is not in the document]
Comment (cjtenny) [reply src/regexp-modifiers/add-remove-multi-duplicate.case:16]: I think it'd be more clear but duplicate a bit of code - I mostly did it this way to hack being able to reuse cases for arithmetic and non-arithmetic cases. I'll make the change! Or someone else feel free to, see my other note (pending)

**Comment by cjtenny** [comment]
A belated thanks Richard for the review - I'm unfortunately on medical leave right now so it'll be another month or so before I can return to this. If anybody else wants to take over it in the meantime, feel free, otherwise I'll pick it up as soon as I'm back.

**Comment by guijemont** [comment]
@gibson042 I've started a new draft PR (https://github.com/tc39/test262/pull/3807) that addresses your comment. Currently a draft PR because the templated test generator does not support the way we format our templates (see https://github.com/tc39/test262/pull/3807#issuecomment-1492571532).

**Comment by ptomato** [comment]
Closing this in favour of #3807. Thanks, @guijemont!

**Pull request closed without merging**